src/url_generator.rs

```diff
@@ -17,20 +22,43 @@ pub enum UrlGeneratorError {
     ParseError(ParseError, String),
     #[error(transparent)]
     FromUtf8Error(#[from] FromUtf8Error),
+    #[error("No valid URLs found")]
+    NoURLsError(),
 }
 
 impl UrlGenerator {
     pub fn new_static(url: Url) -> Self {
         Self::Static(url)
     }
 
+    pub fn new_multi_static(filename: &str) -> io::Result<Self> {
+        let path = Path::new(filename);
+        let file = File::open(path)?;
+        let reader = io::BufReader::new(file);
+
+        let urls: Vec<Url> = reader
+            .lines()
+            .flatten()
+            .map(|url_str| Url::parse(&url_str).unwrap())
+            .collect();
+
+        Ok(Self::MultiStatic(urls))
+    }
+
     pub fn new_dynamic(regex: Regex) -> Self {
         Self::Dynamic(regex)
     }
 
     pub fn generate<R: Rng>(&self, rng: &mut R) -> Result<Cow<Url>, UrlGeneratorError> {
         match self {
             Self::Static(url) => Ok(Cow::Borrowed(url)),
+            Self::MultiStatic(urls) => {
+                if let Some(random_url) = urls.choose(&mut rand::thread_rng()) {
```

Comment:
Please use `rng` argument instead of `thread_rng()`

Reply:
yes, makes sense, done.